pkg/apis/eksctl.io/v1alpha5/defaults.go

```diff
@@ -230,13 +230,22 @@ func getDefaultVolumeType(nodeGroupOnOutposts bool) string {
 	return DefaultNodeVolumeType
 }
 
-func setContainerRuntimeDefault(ng *NodeGroup) {
-	if ng.ContainerRuntime == nil {
-		ng.ContainerRuntime = &DefaultContainerRuntime
+func setContainerRuntimeDefault(ng *NodeGroup, apiVersion string) {
+	var runtime string
+	if ng.ContainerRuntime != nil {
+		return
+	}
+
+	if apiVersion < DockershimDeprecationVersion {
```

Comment:
While it may work in this case, cluster versions should not be compared as strings. They should be parsed and compared. You can use `IsMinVersion` or `CompareVersions` from `utils`.